diff -Naur pystatsd-0.1.7/pystatsd/daemon.py pystatsd-0.1.7.new/pystatsd/daemon.py
--- pystatsd-0.1.7/pystatsd/daemon.py	2011-10-10 11:12:48.000000000 -0700
+++ pystatsd-0.1.7.new/pystatsd/daemon.py	2011-10-10 11:07:40.000000000 -0700
@@ -59,8 +59,11 @@
         # Write the pidfile.
         atexit.register(self.delpid)
         pid = str(os.getpid())
-        with open(self.pidfile, 'w+') as fp:
+	fp = open(self.pidfile, 'w+') 
+        try:
             fp.write('%s\n' % pid)
+        finally:
+            fp.close()
 
     def delpid(self):
         os.remove(self.pidfile)
@@ -69,8 +72,11 @@
         """Start the daemon."""
         pid = None
         if os.path.exists(self.pidfile):
-            with open(self.pidfile, 'r') as fp:
+            fp = open(self.pidfile, 'r') 
+            try: 
                 pid = int(fp.read().strip())
+            finally:
+                fp.close()
 
         if pid:
             msg = 'pidfile (%s) exists. Daemon already running?\n'
@@ -84,8 +90,11 @@
         """Stop the daemon."""
         pid = None
         if os.path.exists(self.pidfile):
-            with open(self.pidfile, 'r') as fp:
+            fp = open(self.pidfile, 'r') 
+            try: 
                 pid = int(fp.read().strip())
+            finally:
+                fp.close()
 
         if not pid:
             msg = 'pidfile (%s) does not exist. Daemon not running?\n'
diff -Naur pystatsd-0.1.7/pystatsd/server.py pystatsd-0.1.7.new/pystatsd/server.py
--- pystatsd-0.1.7/pystatsd/server.py	2011-10-06 17:13:39.000000000 -0700
+++ pystatsd-0.1.7.new/pystatsd/server.py	2011-10-10 11:59:36.000000000 -0700
@@ -107,7 +107,10 @@
         
         for k, v in self.counters.items():
             v = float(v)
-            v = v if self.no_aggregate_counters else v / (self.flush_interval / 1000)
+            if self.no_aggregate_counters:
+	         v = v
+            else:
+                 v / (self.flush_interval / 1000)
 
             if self.debug:
                 print "Sending %s => count=%s" % ( k, v )
